libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1014,6 +1014,30 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 			_memberAccess.expression().accept(*this);
 		return false;
 	}
+	// Another special case for `this.f.selector` which does not need the address.
+	// There are other uses of `.selector` which do need the address, but we want this
+	// specific use to be a pure expression.
+	if (
+		_memberAccess.expression().annotation().type->category() == Type::Category::Function &&
+		member == "selector"
+	)
+		if (auto const* expr = dynamic_cast<MemberAccess const*>(&_memberAccess.expression()))
+			if (auto const* exprInt = dynamic_cast<Identifier const*>(&expr->expression()))
+				if (exprInt->name() == "this")
+					if (Declaration const* declaration = expr->annotation().referencedDeclaration)
+					{
+						u256 identifier;
+						if (auto const* variable = dynamic_cast<VariableDeclaration const*>(declaration))
```

Comment:
Hm I wonder if we actually want to support `.selector` on public variables. I don't think we had test cases for that before.

Reply:
shall I add a test case?

Reply:
It works. I will add a test case.